Apps/WebClient/src/ClientApp/app/components/timeline/commentSection.vue

```diff
@@ -0,0 +1,184 @@
+          <b-form @submit.prevent="addComment">
+            <b-form-input
+              type="text"
+              autofocus
+              class="newComment"
+              v-model="newComment"
+              placeholder="Enter a comment"
+              maxlength="1000"
+            ></b-form-input>
+          </b-form>
+        </b-collapse>
+      </b-col>
+    </b-row>
+    <b-row>
+      <b-col>
+        <b-collapse :visible="commentsVisible">
+          <div v-if="!this.isLoadingComments">
+            <div v-for="comment in this.comments" :key="comment.id">
+              <Comment :comment="comment"></Comment>
+            </div>
+          </div>
+        </b-collapse>
+      </b-col>
+    </b-row>
+  </div>
+</template>
+<script lang="ts">
+import Vue from "vue";
+import UserComment from "@/models/userComment";
+import CommentComponent from "@/components/timeline/comment.vue";
+import MedicationTimelineEntry from "@/models/medicationTimelineEntry";
+import User from "@/models/user";
+import { Prop, Component } from "vue-property-decorator";
+import { IUserCommentService } from "@/services/interfaces";
+import { SERVICE_IDENTIFIER } from "@/plugins/inversify";
+import container from "@/plugins/inversify.config";
+import {
+  faCommentAlt,
+  IconDefinition,
+} from "@fortawesome/free-solid-svg-icons";
+import { Getter } from "vuex-class";
+
+@Component({
+  components: {
+    Comment: CommentComponent,
+  },
+})
+export default class CommentSectionComponent extends Vue {
+  @Getter("user", { namespace: "user" }) user!: User;
+  @Prop() parentEntry!: MedicationTimelineEntry;
+  private commentService!: IUserCommentService;
+  private commentsVisible: boolean = false;
+  private commentInputVisible: boolean = false;
+  private isCommentSaving: boolean = false;
+  private isLoadingComments: boolean = false;
+  private newComment: string = "";
+  private comments: UserComment[] = [];
+  private numComments = 0;
+  private hasErrors: boolean = false;
+
+  private mounted() {
+    this.commentService = container.get<IUserCommentService>(
+      SERVICE_IDENTIFIER.UserCommentService
+    );
+    this.getComments();
+  }
+
+  private get hasComments(): boolean {
+    return this.comments.length > 0;
+  }
+
+  private get commentsLoaded(): boolean {
+    return this.commentsLoaded;
+  }
+
+  private get commentIcon(): IconDefinition {
+    return faCommentAlt;
+  }
+
+  private sortComments() {
+    this.comments.sort((a, b) => {
+      if (a.createdDateTime > b.createdDateTime) {
+        return -1;
+      } else if (a.createdDateTime < b.createdDateTime) {
+        return 1;
+      } else {
+        return 0;
+      }
+    });
+  }
+
+  private toggleComments(): void {
+    this.commentsVisible = !this.commentsVisible;
+    if (this.commentInputVisible && !this.commentsVisible) {
+      this.toggleCommentInput();
+    }
+  }
+
+  private toggleCommentInput(): void {
+    this.commentInputVisible = !this.commentInputVisible;
```

Comment:
this method and toggleComments seem to mirror eachother. Should one just call the other?

Reply:
Played around with that idea and the behaviour was weird. This was the most predictable behaviour I could come up with. Otherwise the comment input persists after collapsing all comments.